### ISSUE TITLE
Annual increases

### ISSUE THREAD

**Issue by hunkyn** [microrealestate/microrealestate#209]
Need an option to allow for annual rent increases. Have a multi unit commercial property where there is annual escalator in rent that changes based on CPI.

**Comment by mykael90** [comment]
I am also interested in this issue. Annual rent adjustments are a common market practice.
I briefly analyzed how this could be implemented and one suggestion that would take advantage of the business logic already used was to transform the expense entry in the LeaseContractForm into an array entry to insert multiple expenses and attach the start and end base date to define which terms would be applied. In the 1_base.js task, when computing the terms, if it was within the date condition, the expense would be included. In the database schema, the "expenses" field in >occupants>properties is already defined as an array. It is an array, but at the moment the user can only insert a single expense from what I understand.
Finally, to insert a correction in the rent amount, the user would edit the contract and insert a new expense with the appropriate start date and the end date coinciding with the end date of the contract. The following year, to correct it again, it would be enough to insert a new expense and so on. The total amount would be defined by the sum.
Of course this would be a more practical option, I understand that the ideal would be to have a logic created only for annual (or other) correction.

**Comment by camelaissani** [comment]
@mykael90 good "brief" investigation 👍

**Comment by camelaissani** [comment]
@mykael90 the solution you described might solve #175

**Comment by mykael90** [comment]
That's true. Implementing this functionality could also temporarily address the issue of annual adjustments. That would be great.
